### PR TITLE
warn user about canvas anti-fingerprinting issues

### LINKED ISSUE #399: Brave device recognition blocked setting causes reticketing to fail
## Context

Browser: Brave Version 1.3.118 Chromium: 80.0.3987.116 (Official Build) (64-bit)
MacOS: 10.15.3
Logged into Bridge via Metamask (and mnemonic, same results)
Click Admin
Click Reticket
The "I understand" button on the first of four pages won't click—nothing happens
Click logout

NOTE: web inspection of the reticket page 1 of 4 after unsuccessfully clicking "I understand" shows:
```
Uncaught (in promise) DOMException: Failed to execute 'atob' on 'Window': The string to be decoded is not correctly encoded.
    at dataURItoBlob (https://bridge.urbit.org/static/js/2.0afadec5.chunk.js:118446:22)
    at https://bridge.urbit.org/static/js/2.0afadec5.chunk.js:154267:18
    at Array.map (<anonymous>)
    at generatePng (https://bridge.urbit.org/static/js/2.0afadec5.chunk.js:154197:38)
    at https://bridge.urbit.org/static/js/2.0afadec5.chunk.js:154343:28
    at Array.map (<anonymous>)
    at PaperRenderer._callee2$ (https://bridge.urbit.org/static/js/2.0afadec5.chunk.js:154341:39)
    at tryCatch (https://bridge.urbit.org/static/js/2.0afadec5.chunk.js:172631:17)
    at Generator.invoke [as _invoke] (https://bridge.urbit.org/static/js/2.0afadec5.chunk.js:172850:22)
    at Generator.prototype.<computed> [as next] (https://bridge.urbit.org/static/js/2.0afadec5.chunk.js:172684:21)
```

## Error

```
TypeError: Cannot read property 'data' of undefined
    at https://bridge.urbit.org/static/js/main.4b7d939e.chunk.js:1194:911
    at Object.useMemo (https://bridge.urbit.org/static/js/2.0afadec5.chunk.js:159329:9)
    at useMemo (https://bridge.urbit.org/static/js/2.0afadec5.chunk.js:155544:16)
    at useRouter (https://bridge.urbit.org/static/js/main.4b7d939e.chunk.js:1194:862)
    at Admin (https://bridge.urbit.org/static/js/main.4b7d939e.chunk.js:1880:1134)
    at dh (https://bridge.urbit.org/static/js/2.0afadec5.chunk.js:159000:7)
    at Jh (https://bridge.urbit.org/static/js/2.0afadec5.chunk.js:159476:7)
    at mj (https://bridge.urbit.org/static/js/2.0afadec5.chunk.js:161749:86)
    at jj (https://bridge.urbit.org/static/js/2.0afadec5.chunk.js:160737:11)
    at Z (https://bridge.urbit.org/static/js/2.0afadec5.chunk.js:160594:15)
```

In Bridge version 2.3.3.

### PR DESCRIPTION
**Pull request by liam-fitzgerald**
fixes #399 #374

warns user if we cannot render to canvas properly
Screen Shot 2020-03-12 at 5 38 22 pm

copy is rubbish, any help?

### CODE REVIEW

File: src/components/CanvasSupportWarning.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            Something is blocking your web browser from rendering your wallets.
+            Please turn off any anti-fingerprinting extensions and try again.
+            <br />
+            <br />
+            Brave users: Please set the device recognition setting in the brave
+            shield menu to "Cross-site device recognition blocked"
+            <br />
+            <br />
+            Firefox users: Please click on the icon of an image in your URL bar
+            and allow bridge.urbit.org to use your HTML5 canvas image data
```
Comment (Fang-) [inline src/components/CanvasSupportWarning.js]: Might want these as `<P>` instead.

As for copy, can we not detect the browser based on useragent string (or even the way the check fails) and display just the appropriate message accordingly? (We'll want periods at the end of these sentences either way.)

Maybe we can include small screenshots to help point them at the right thing?

Do they need to reload, or does the change take effect immediately?

Comment (liam-fitzgerald) [reply src/components/CanvasSupportWarning.js]: they just need to click the retry button and it will recheck support.

**Comment by liam-fitzgerald** [comment]
@jyng do you think the workarounds should go into bridge itself, or in the docs?

**Comment by jyng** [comment]
Yeah, I feel like the latter bit concerning different browsers should go into docs. There should be a link here that says "Learn How"

File: src/components/CanvasSupportWarning.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+  return (
+    <Flex col className={className}>
+      <canvas ref={ref} style={{ display: 'none ' }} />
+      {!supported.getOrElse(true) && (
+        <>
+          <Flex.Item as={WarningBox} className={cn(warningClassName, 'f6 mb4')}>
+            Something is blocking your web browser from rendering your wallets.
+            <br />
+            <LinkButton href="https://urbit.org/using/bridge-troubleshooting">
```
Comment (Fang-) [inline src/components/CanvasSupportWarning.js]:
```suggestion
            <LinkButton href="https://urbit.org/using/operations/bridge-troubleshooting/">
```

**Comment by Fang-** [comment]
ping @jyng for approval review (if you think this is ready) and @liam-fitzgerald for merge conflicts

**Comment by jyng** [comment]
My suggestion:

Error: Your wallet was not rendered because of a web browser issue. 

<LinkButton> Learn More </LinkButton>

**Comment by Fang-** [comment]
ping @liam-fitzgerald

**Pull request closed without merging**